meilisearch/config.py

```diff
@@ -42,7 +42,7 @@ def __init__(
         self,
         url: str,
         api_key: Optional[str] = None,
-        timeout: Optional[int] = None,
+        timeout: Optional[Union[Union[int, float], Tuple[Union[int, float]]]] = None,
```

Comment:
This can be simplified. Additionally the tuples as they are defined will fail mypy, they should be `Tuple[int, int]` and `Tuple[float, float]`.

```suggestion
        timeout: Optional[Union[int, float, Tuple[int, int], Tuple[float, float]]] = None,
```

Reply:
I just realized you may have been going for `timeout: Optional[Union[int, float, Tuple[Union[int, float], Union[int, float]]]]`. If so this is fine as long as mypy allows it. There are some restrictions on tuple types, and I haven’t tried one like this before so I’m not sure if it will be allowed. If it is we would also need to add `(5, 5.0)` and `(5.0, 5)` to the test parameters.

Reply:
I verified mypy is fine with `Optional[Union[int, float, Tuple[Union[int, float], Union[int, float]]]]` so we can go with this one instead of `Optional[Union[int, float, Tuple[int, int], Tuple[float, float]]]`

```suggestion
        timeout: Optional[Union[int, float, Tuple[Union[int, float], Union[int, float]]]] = None,
````